applications/RANSModellingApplication/custom_processes/auxiliary_processes/rans_vector_align_process.h

```diff
@@ -0,0 +1,327 @@
+    using NodeType = ModelPart::NodeType;
+
+    /// Pointer definition of RansVectorAlignProcess
+    KRATOS_CLASS_POINTER_DEFINITION(RansVectorAlignProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor
+
+    RansVectorAlignProcess(Model& rModel, Parameters& rParameters)
+        : mrModel(rModel), mrParameters(rParameters)
+    {
+        KRATOS_TRY
+
+        Parameters default_parameters = Parameters(R"(
+        {
+            "model_part_name"         : "PLEASE_SPECIFY_MODEL_PART_NAME",
+            "input_variable_name"     : "PLEASE_SPECIFY_INPUT_VARIABLE_NAME",
+            "output_variable_name"    : "PLEASE_SPECIFY_OUTPUT_VARIABLE_NAME",
+            "alignment_variable_name" : "NORMAL",
+            "is_tangential_alignment" : true,
+            "echo_level"              : 0
+        })");
+
+        mrParameters.ValidateAndAssignDefaults(default_parameters);
+
+        mModelPartName = mrParameters["model_part_name"].GetString();
+        mInputVariableName = mrParameters["input_variable_name"].GetString();
+        mOutputVariableName = mrParameters["output_variable_name"].GetString();
+        mAlignmentVariableName = mrParameters["alignment_variable_name"].GetString();
+        mIsTangentialAlignment = mrParameters["is_tangential_alignment"].GetBool();
+        mEchoLevel = mrParameters["echo_level"].GetInt();
+
+        KRATOS_CATCH("");
+    }
+
+    /// Destructor.
+    ~RansVectorAlignProcess() override
+    {
+        // delete mpDistanceCalculator;
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    int Check() override
+    {
+        return 0;
+    }
+
+    void Execute() override
+    {
+        CalculateAlignmentVectors();
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return std::string("RansVectorAlignProcess");
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << this->Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    Model& mrModel;
+    Parameters& mrParameters;
```

Comment:
parametes should be passed by value. The managing of the resources is done internally

Reply:
done